demo/Dockerfile.dev

```diff
@@ -0,0 +1,14 @@
+FROM node:8.16.0-stretch AS texture
+
+ARG UNAME=testuser
```

Comment:
(For development only)
Trying this out as a way to run the container user the same as the host user, so that created files that are mounted to the host are owned by the host user. Otherwise they are usually owned by root or require having to give write permissions to everyone.

See [Stackoverflow](https://stackoverflow.com/a/44683248/8676953). Although he is also referring to [entrypoint.sh magic](https://github.com/sudo-bmitch/jenkins-docker/blob/master/entrypoint.sh) which I tried before but seemed more complex and didn't quite work as well.

Reply:
The entrypoint magic is what I think of as the idiomatic solution, but I am not sure there is a standard implementation of it (`gosu`?)

Reply:
The downside is that the image generated by the build changes depending on the build argument, which makes it only viable for development rather than publishing it.

Reply:
Yes, although it has reasonable defaults (assuming they don't collide with the `node` user). This image is inherently designed for development. It doesn't include the actual build beyond providing the "development environment". Although for the purpose, we could use a published image as well in theory. I am happy to use whatever works and requires the least manual steps by the developer.

For me, using the entrypoint magic resulted in my username with a suffix. But I haven't properly tested it I think.